samples/remote_attestation/common/attestation.cpp

```diff
@@ -2,163 +2,195 @@
- * accompanying data is ensured by comparing its SHA256 digest against the
- * report_data field.
+ * 1) The remote evidence is first attested using the oe_verify_evidence API.
+ * This ensures the authenticity of the enclave that generated the remote
+ * evidence. 2) Next, to establish trust of the enclave that  generated the
+ * remote evidence, the mrsigner, product_id, isvsvn values are checked to  see
+ * if they are predefined trusted values.
  */
-bool Attestation::attest_remote_report(
-    const uint8_t* remote_report,
-    size_t remote_report_size,
+bool Attestation::attest_remote_evidence(
+    const uint8_t* remote_evidence,
+    size_t remote_evidence_size,
     const uint8_t* data,
     size_t data_size)
 {
     bool ret = false;
     uint8_t sha256[32];
-    oe_report_t parsed_report = {0};
     oe_result_t result = OE_OK;
+    oe_result_t verifier_result = OE_OK;
+    oe_claim_t* claims = NULL;
+    size_t claims_length = 0;
 
-    // While attesting, the remote report being attested must not be tampered
+    // While attesting, the remote evidence being attested must not be tampered
     // with. Ensure that it has been copied over to the enclave.
-    if (!oe_is_within_enclave(remote_report, remote_report_size))
+    if (!oe_is_within_enclave(remote_evidence, remote_evidence_size))
     {
-        TRACE_ENCLAVE("Cannot attest remote report in host memory. Unsafe.");
+        TRACE_ENCLAVE("Cannot attest remote evidence in host memory. Unsafe.");
         goto exit;
     }
 
-    // 1)  Validate the report's trustworthiness
-    // Verify the remote report to ensure its authenticity.
-    result =
-        oe_verify_report(remote_report, remote_report_size, &parsed_report);
-    if (result != OE_OK)
+    // Initialize the verifier.
+    verifier_result = oe_verifier_initialize();
+    if (verifier_result != OE_OK)
     {
-        TRACE_ENCLAVE("oe_verify_report failed (%s).\n", oe_result_str(result));
+        TRACE_ENCLAVE("oe_verifier_initialize failed.");
         goto exit;
     }
 
-    // 2) validate the enclave identity's signed_id is the hash of the public
-    // signing key that was used to sign an enclave. Check that the enclave was
-    // signed by an trusted entity.
-    if (memcmp(parsed_report.identity.signer_id, m_enclave_mrsigner, 32) != 0)
+    // Validate the evidence's trustworthiness
+    // Verify the remote evidence to ensure its authenticity.
+    result = oe_verify_evidence(
+        &sgx_remote_uuid,
+        remote_evidence,
+        remote_evidence_size,
+        NULL,
+        0,
+        NULL,
+        0,
+        &claims,
+        &claims_length);
+    if (result != OE_OK)
     {
-        TRACE_ENCLAVE("identity.signer_id checking failed.");
         TRACE_ENCLAVE(
-            "identity.signer_id %s", parsed_report.identity.signer_id);
+            "oe_verify_evidence failed (%s).\n", oe_result_str(result));
+        goto exit;
+    }
+
+    uint8_t* value;
```

Comment:
The convention is to declare all local variables at the beginning of the function.

Reply:
Fixed.